web/src/compose_ui.ts

```diff
@@ -24,8 +24,8 @@ import * as util from "./util";
 
 // TODO: Refactor to push this into a field of ComposeTriggeredOptions.
 type messageType = "stream" | "private";
-type ComposeTriggeredOptions = {
-    trigger: string;
+export type ComposeTriggeredOptions = {
+    trigger?: string;
```

Comment:
This change seems wrong; I think instead this function should add a `trigger: "switch_message_type"` in the object it constructs.
```
function switch_message_type(message_type) {                                                            
    $("#compose-content .alert").hide();                                                                
                                                                                                        
    compose_state.set_message_type(message_type);                                                       
                                                                                                        
    const opts = {                                                                                      
        message_type,                                                                                   
        stream_id: compose_state.stream_id(),                                                           
        topic: compose_state.topic(),                                                                   
        private_message_recipient: compose_state.private_message_recipient(),                           
    };                                                                                                  
    update_compose_for_message_type(message_type, opts);                                                
    update_placeholder_text();                                                                          
    compose_ui.set_focus(message_type, opts);                                                           
}                                                                                                       
```